collect_app/src/test/java/org/odk/collect/android/widgets/SelectOneWidgetTest.java

```diff
@@ -104,29 +106,21 @@ public void whenChoicesHaveAudio_logsAudioChoiceEvent() throws Exception {
                 ))
                 .build();
 
-        populateRecyclerView(getActualWidget());
+        populateRecyclerView(getWidget());
         verify(analytics).logEvent("Prompt", "AudioChoice", "formAnalyticsID");
     }
 
-    private void overrideDependencyModule() throws Exception {
-        ReferenceManager referenceManager = setupFakeReferenceManager(REFERENCES);
-        RobolectricHelpers.overrideAppDependencyModule(new AppDependencyModule() {
-
-            @Override
-            public ReferenceManager providesReferenceManager() {
-                return referenceManager;
-            }
-
-            @Override
-            public AudioHelperFactory providesAudioHelperFactory() {
-                return context -> audioHelper;
-            }
+    @Test
+    public void whenAChoiceIsBlank_itIsShownAsBlank() {
```

Comment:
I don't understand this test! The test that is introduced should fail or crash at this point and this one passes. I don't understand "shown" in its name. This issue is only about the underlying value, not about what is displayed, right? If you were going to test what is displayed, `innerText` on `getChildAt(0)` wouldn't make sense because that gives you the linear layout that contains the audio-video-image-text-label view.

I think you probably want to do something like select the first choice and verify that its value is blank. Again, it should fail or crash in this commit.

Reply:
I see from looking at the other commit that you seem to have intentionally introduced a test for a blank value that was intended to pass.

Reply:
Yeah I wanted to check the blank and `null` cases. It's calling `widget.getChoicesList().getChildAt(0)` so it is checking what is displayed for that select choice. That being said I agree that isn't really the right test here. I think when I wrote this (a wee while ago) I confused value and label. As you say the "feature" we're actually trying to drive out here is that when you select a blank/`null` value you get a blank/`null` answer in the widget. Rest assured a forehead has been slapped and this will be changed.